components/SEOBundle/bundle/Controller/SitemapController.php

```diff
@@ -214,4 +252,157 @@ protected function fillSitemapIndex(DOMDocument $sitemap, int $numberOfResults,
+                        ['page' => $sitemapNumber]
+                    );
+                    $loc = $sitemap->createElement('loc', $locUrl);
+                    $date = new DateTime();
+                    $modificationDate = $date->format('c');
+                    $mod = $sitemap->createElement('lastmod', $modificationDate);
+                    $sitemapElt->appendChild($loc);
+                    $sitemapElt->appendChild($mod);
+                    $root->appendChild($sitemapElt);
+                }
+            }
+        });
+    }
+
+    /**
+     * Fill a sitemap.
+     */
+    protected function fillMultiLanguagesSitemap(
+        DOMDocument $sitemap,
+        DOMElement $root,
+        SearchResult $results
+    ): void {
+        $currentSiteAccess = $this->siteMapHelper->getCurrentSiteAccess();
+        foreach ($results->searchHits as $searchHit) {
+            /**
+             * @var Location  $location
+             */
+            $location = $searchHit->valueObject;
+
+            $mainLanguageUrl = $this->siteMapHelper->generateLocationUrl($location->id, $currentSiteAccess);
+            if (null === $mainLanguageUrl || 0 != strpos($mainLanguageUrl, 'view/content/')) {
+                continue;
+            }
+
+            $modified = $location->contentInfo->modificationDate->format('c');
+            $loc = $sitemap->createElement('loc', $mainLanguageUrl);
+            $lastmod = $sitemap->createElement('lastmod', $modified);
+            $urlElt = $sitemap->createElement('url');
+
+            $urlElt->appendChild($loc);
+            // Inject the image tags if config is enabled
+            $this->injectImageTag($location, $sitemap, $urlElt);
+            // Inject the alternate lang tags if config is enabled
+            $this->injectAlternateLangTag($location, $sitemap, $urlElt);
+
+            $urlElt->appendChild($lastmod);
+            $root->appendChild($urlElt);
+        }
+    }
+
+    public function injectAlternateLangTag($location, DOMDocument $sitemap, DOMElement $root): void
+    {
+        $isMultiLanguages = $this->getConfigResolver()->getParameter('multi_languages_sitemap', 'nova_ezseo');
+        if ($isMultiLanguages) {
+            try {
+                $siteAccesses = $this->getConfigResolver()->getParameter('translation_siteaccesses');
+                $languagesCodes = [];
+                $contentInfo = $location->contentInfo;
+                $contentLanguages = $this->repository->getContentService()
+                    ->loadVersionInfo($contentInfo)->getLanguages();
+                foreach ($contentLanguages as $language) {
+                    $languagesCodes[] = $language->languageCode;
+                }
+                foreach ($siteAccesses as $siteAccess) {
+                    $siteAccessMainLanguage = $this->siteMapHelper->getSiteAccessMainLanguage($siteAccess);
+                    if (!in_array($this->siteMapHelper->getSiteAccessMainLanguage($siteAccess), $languagesCodes)) {
```

Comment:
Tu appel deux fois la fonction `$this->siteMapHelper->getSiteAccessMainLanguage($siteAccess)` au lieu d'utiliser la variable existante`!in_array($siteAccessMainLanguage, $languagesCodes)`

Reply:
c'est réglé